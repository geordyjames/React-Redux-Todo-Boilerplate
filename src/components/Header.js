import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';
import logo from '../images/logo.png';


export const Header = ({ startLogout }) => (
	<nav className="uk-navbar-container uk-navbar-transparent header" data-uk-navbar>
		<div className="uk-navbar-left  uk-logo">
			<Link
        className="uk-navbar-item uk-logo uk-text-bold header__logo"
				to="/"
			>
				<img src={logo} alt="Logo" />
				React Redux Todo Boilerplate
			</Link>
		</div>
		<div className="uk-navbar-right">
			<div className="uk-navbar-item">
				<button className="uk-button uk-button-default" onClick={startLogout}>
          <span data-uk-icon="sign-out"></span> Logout
				</button>
			</div>
		</div>
	</nav>
);

const mapDispatchToProps = dispatch => ({
	startLogout: () => dispatch(startLogout())
});

export default connect(
	undefined,
	mapDispatchToProps
)(Header);
