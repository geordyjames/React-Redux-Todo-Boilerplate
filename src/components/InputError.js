import React from 'react';

const InputError = ({error}) => (
  <div className="uk-alert-danger" data-uk-alert>
    <p> {error} </p>
  </div>
);

export default InputError;