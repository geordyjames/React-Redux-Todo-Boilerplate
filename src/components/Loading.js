import React from 'react';

const Loading = props => {
	if (props.error) {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle half-height">
				Error!
				<button
					className="uk-button uk-button-danger uk-text-bold uk-margin-left"
					onClick={props.retry}
				>
					Retry
				</button>
			</div>
		);
	} else if (props.timedOut) {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle half-height">
				Taking a long time...
				<button
					className="uk-button uk-button-danger uk-text-bold uk-margin-left"
					onClick={props.retry}
				>
					Retry
				</button>
			</div>
		);
	} else if (props.pastDelay) {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle half-height">
				<div data-uk-spinner="ratio: 3" />
			</div>
		);
	} else {
		return null;
	}
};

export default Loading;
