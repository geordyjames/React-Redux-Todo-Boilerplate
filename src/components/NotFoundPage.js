import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
	<div
		className="uk-flex uk-flex-center uk-flex-middle"
		style={{ height: '100vh' }}
	>
		404 -{' '}
		<Link className="uk-button uk-button-default" to="/">
			Go home
		</Link>
	</div>
);

export default NotFoundPage;
