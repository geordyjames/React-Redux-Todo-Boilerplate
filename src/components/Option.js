import React from 'react';

const Option = props => (
	<li>
		<p className="uk-text-middle uk-align-left uk-margin-small-bottom">
			{props.count}. {props.optionText}
		</p>
		<button
			className="uk-icon-button uk-align-right"
			data-uk-icon="icon: trash"
			onClick={() => {
				props.handleDeleteOption(props.id);
			}}
		/>
	</li>
);

export default Option;
