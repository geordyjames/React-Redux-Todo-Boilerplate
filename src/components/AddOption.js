import React from 'react';
import { connect } from 'react-redux';
import { add } from '../actions/todo';
import InputError from './InputError';

export class AddOption extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			error: null
		};

		this.handleAddOption = this.handleAddOption.bind(this);
	}

	async handleAddOption(e) {
		e.preventDefault();
		e.persist();
		const option = e.target.elements.option.value.trim();

		if (!option) {
			await Promise.resolve(
				this.setState(() => ({ error: 'Enter valid value to add item' }))
			);
		} else if (this.props.todo.find(todo => todo.name === option)) {
			await Promise.resolve(
				this.setState(() => ({ error: 'This option already exists' }))
			);
		} else {
			await Promise.resolve(this.setState(() => ({ error: null })));
			this.props.handleAddOption(option);
			e.target.elements.option.value = '';
		}
	}

	render() {
		return (
			<form className="uk-margin" onSubmit={this.handleAddOption}>
				<input
					className="uk-input uk-form-width-medium"
					type="text"
					name="option"
					placeholder="Enter a value"
				/>
				<button className="uk-button uk-button-primary uk-margin-left uk-text-bold">
					<span data-uk-icon="plus" /> Add Option
				</button>
				{this.state.error && <InputError error={this.state.error} />}
			</form>
		);
	}
}

const mapStateToProps = state => ({
	todo: state.todo
});

const mapDispatchToProps = dispatch => ({
	handleAddOption: option => dispatch(add(option))
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(AddOption);
