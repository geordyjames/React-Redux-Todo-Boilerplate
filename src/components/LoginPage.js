import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import InputError from './InputError';
import logo from '../images/logo.png';

export class LoginPage extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			error: null
		};

		this.startLogin = this.startLogin.bind(this);
	}

	startLogin(e) {
		e.preventDefault();
		const name = e.target.elements.name.value.trim();

		if (!name) {
			this.setState(() => ({ error: 'Enter your name to login' }));
		} else {
			this.setState(() => ({ error: null }));
			this.props.startLogin(name);
		}
	}

	render() {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle full-height">
				<div className="uk-text-center">
					<img src={logo} alt="Logo" />
					<h1 className="uk-heading-hero">React Redux Todo Boilerplate</h1>
					<form className="uk-padding" onSubmit={this.startLogin}>
						<input
							className="uk-input uk-form-width-medium"
							type="text"
							name="name"
							placeholder="Enter your name"
						/>
						<button className="uk-button uk-button-primary uk-margin-left uk-text-bold">
							<span data-uk-icon="sign-in" /> Login
						</button>
						{this.state.error && <InputError error={this.state.error} />}
					</form>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	startLogin: name => dispatch(startLogin(name))
});

export default connect(
	undefined,
	mapDispatchToProps
)(LoginPage);
