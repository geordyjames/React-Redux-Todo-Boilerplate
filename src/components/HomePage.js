import React from 'react';
import { connect } from 'react-redux';
import { remove } from '../actions/todo';
import AddOption from './AddOption';
import Options from './Options';

export const HomePage = props => (
	<div className="uk-container">
		<h1 className="uk-heading-primary uk-text-center uk-padding ">
			{props.name}'s Todo List
		</h1>
		<div className="uk-text-center">
			<AddOption />
		</div>
		<div className="uk-flex uk-flex-center">
			<Options
				options={props.todo}
				handleDeleteOption={props.handleDeleteOption}
			/>
		</div>
	</div>
);

const mapStateToProps = state => ({
	name: state.auth.name,
	todo: state.todo
});

const mapDispatchToProps = dispatch => ({
	handleDeleteOption: id => dispatch(remove(id))
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(HomePage);
