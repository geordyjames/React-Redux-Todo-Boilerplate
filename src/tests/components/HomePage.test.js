import React from 'react';
import { shallow } from 'enzyme';
import todos from '../fixtures/todos';
import { HomePage } from '../../components/HomePage';

test('should render HomePage correctly', () => {
  const wrapper = shallow(<HomePage name="Hello" todo={todos} handleDeleteOption={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});