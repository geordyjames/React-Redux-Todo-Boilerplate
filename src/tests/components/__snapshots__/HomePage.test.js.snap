// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage correctly 1`] = `
<div
  className="uk-container"
>
  <h1
    className="uk-heading-primary uk-text-center uk-padding "
  >
    Hello
    's Todo List
  </h1>
  <div
    className="uk-text-center"
  >
    <Connect(AddOption) />
  </div>
  <div
    className="uk-flex uk-flex-center"
  >
    <Options
      handleDeleteOption={[Function]}
      options={
        Array [
          Object {
            "id": "1",
            "name": "Gum",
          },
          Object {
            "id": "2",
            "name": "Rent",
          },
          Object {
            "id": "3",
            "name": "Credit Card",
          },
        ]
      }
    />
  </div>
</div>
`;
